.github/workflows/precommit.yaml

```diff
@@ -0,0 +1,47 @@
+name: "Precommit tests"
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/precommit.yaml'
+      - 'llvm/**'
+
```

Comment:
Do we need to define concurrency groups?  What's the default behavior if a PR gets updated while its previous version is still being tested?

Reply:
Good point. It currently just keeps running multiple jobs and won't cancel the out of date ones. I've added the default concurrency group associated with the PR to fix this.